src/peft/utils/save_and_load.py

```diff
@@ -184,6 +184,27 @@ def renamed_dora_weights(k):
         to_return = {k: state_dict[k] for k in state_dict if "internal_xlora_classifier" in k}
     elif config.peft_type == PeftType.HRA:
         to_return = {k: state_dict[k] for k in state_dict if "hra_" in k}
+    elif config.peft_type == PeftType.VBLORA:
+        to_return = {}
+        if config.num_vectors < 2**8:
+            indices_dtype = torch.uint8
+        elif config.num_vectors < 2**15:
+            indices_dtype = torch.int16
+        elif config.num_vectors < 2**31:
+            indices_dtype = torch.int32
+        else:
+            indices_dtype = torch.int64
```

Comment:
How much of a difference does this really make? Could this be omitted to make the code simpler?

Reply:
This would be useful when aiming for "extreme" parameter efficiency and aligns with the claims made in the paper.

Reply:
Okay, is there some measurement how this impacts size? Maybe it's in the paper and I missed it.

Reply:
Sure, let's take the Llama2 7B model as an example.

We use a vector bank of 2,048 vectors, each with a length of 256. The rank is set to 4, and K is set to 2, applied to all linear layers. 2,048 vectors can be indexed with an int16.

1. Using int16: The float32-equivalent number of parameters is 680,448, i.e., 2.59 MB.
2. Using int32: The float32-equivalent number of parameters is 758,258, i.e., 2.89 MB. (+11.6% compare to int16)
3. Using int64 (default dtype for index): The float32-equivalent number of parameters is 914,688, i.e., 3.49 MB. (+34.7% compare to int16).

In fact, I believe int16 is sufficient for most cases since it can support up to 32,768 vectors (assuming we don't use negative numbers for indexing). However, I considered all possible cases to ensure robustness. I'd like to know your opinion.